ft-sdk/src/auth/session.rs

```diff
@@ -4,56 +4,272 @@ pub struct SessionID(pub String);
+        .domain(host.without_port())
+        .path("/")
+        .max_age(max_age)
+        .same_site(cookie::SameSite::Strict)
+        .build();
+
+    // Convert the cookie to an HTTP header value and return it
+    Ok(http::HeaderValue::from_str(cookie.to_string().as_str())?)
+}
+
+/// Expires the session cookie immediately by setting its expiration time to the current time.
+#[cfg(feature = "auth-provider")]
+pub fn expire_session_cookie(host: ft_sdk::Host) -> Result<http::HeaderValue, ft_sdk::Error> {
+    let cookie = cookie::Cookie::build((ft_sdk::auth::SESSION_KEY, ""))
+        .domain(host.without_port())
+        .path("/")
+        .expires(convert_now_to_offsetdatetime())
+        .build();
+
+    Ok(http::HeaderValue::from_str(cookie.to_string().as_str())?)
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum SessionIDError {
+    #[error("session not found")]
+    SessionNotFound,
+    #[error("session expired")]
+    SessionExpired(String),
+    #[error("failed to query db: {0:?}")]
+    DatabaseError(#[from] diesel::result::Error),
+}
+
+impl SessionID {
+    /// Retrieves a session ID based on a given user ID.
+    ///
+    /// This method queries the database to find a session associated with the
+    /// given user ID.
+    pub fn from_user_id(
+        conn: &mut ft_sdk::Connection,
+        user_id: &ft_sdk::auth::UserId,
+    ) -> Result<SessionID, SessionIDError> {
+        use diesel::prelude::*;
+        use ft_sdk::auth::fastn_session;
+
+        // Get the current time.
+        let now = ft_sdk::env::now();
+
+        // Query to find the session ID and its expiration time for the given user ID.
+        match fastn_session::table
+            .select((fastn_session::id, fastn_session::expires_at.nullable()))
+            .filter(fastn_session::uid.eq(user_id.0))
+            .first::<(String, Option<chrono::DateTime<chrono::Utc>>)>(conn)
+        {
+            // If a session is found and it is expired, return a `SessionExpired` error.
+            Ok((id, Some(expires_at))) if expires_at < now => {
+                return Err(SessionIDError::SessionExpired(id))
+            }
+            // If a valid session is found, return the session ID.
+            Ok((id, _)) => Ok(SessionID(id)),
+            // If no session is found for the user ID, return a `SessionNotFound` error.
+            Err(diesel::NotFound) => return Err(SessionIDError::SessionNotFound),
+            // If any other error occurs during the query, return it.
+            Err(e) => return Err(e.into()),
+        }
+    }
+
+    /// Validates if the session is active and not expired.
+    ///
+    /// This function checks the validity of the session associated with the session ID.
+    /// If the session is found and is not expired, it returns `Ok(())`.
+    /// If the session is expired or not found, it returns an appropriate error.
+    pub fn validate_session(&self, conn: &mut ft_sdk::Connection) -> Result<(), SessionIDError> {
```

Comment:
Nopes. If session id cookie exists, and session is present in DB it is valid. We dont need this.

Reply:
We have added `expires_at` column in the `fastn_session` table. So if the session expires it becomes invalid.

Reply:
Let's remove session expiry related stuff from here, this will need more thought. We need multiple behaviours, and let developer opt in to their desired behaviour related to session expiry.